src/compiler/checker.ts

```diff
@@ -10403,6 +10433,7 @@ namespace ts {
                 case SyntaxKind.TemplateExpression:
                     return checkTemplateExpression(<TemplateExpression>node);
                 case SyntaxKind.StringLiteral:
+                    return checkStringLiteralExpression(<LiteralExpression>node);
                 case SyntaxKind.NoSubstitutionTemplateLiteral:

```

Comment:
And with string enums, it gets a little better because the declarations merge:
```
const enum MSEvents {
   MSPointerCancel = "MSPointerCancel",
   MSPointerDown,
   MSPointerEnter
}
const enum MSEvents {
   MSOtherThing
}
addEventListener(type: MSEvents, listener: (ev: MSPointerEvent) => any, useCapture?: boolean): void;
```